src/components/SharedComponents/PermissionGateContainer.tsx

```diff
@@ -204,6 +206,32 @@ const PermissionGateContainer = ( {
     }
   }, [result, children, setModalShown] );
 
+  useEffect( () => {
+  // We need to handle permission changes manually on Android
+  // Permissions modified from the settings are not captured automatically
+  // So we check permissions when the app is transitioned from background to foreground
+  // This check is performed only when the platform is Android and the permission result is BLOCKED
+    if ( Platform.OS === "android"
+       && ( result === RESULTS.BLOCKED ) ) {
+      const onAppStateChange = async ( nextAppState: AppStateStatus ) => {
+        if ( prevAppState.current.match( /inactive|background/ ) && nextAppState === "active" ) {
+          await checkPermission();
+        }
+
+        prevAppState.current = nextAppState;
+      };
+
+      const subscription = AppState.addEventListener( "change", onAppStateChange );
+
+      return () => {
+        subscription?.remove();
+      };
+    }
+
+    return () => undefined;
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why are you ignoring the exhaustive deps warning?

Reply:
added checkPermission dependency